wordpress-mobile/WordPress-iOS
# Add universal link action for the Jetpack overlay

Refs p1667832135185539-slack-C03SW66LAER

This fixes the issue where the "Try the new Jetpack app" doesn't point to the Jetpack app. With this PR, two scenarios will be handled:

• | Scenario 1 | Scenario 2
-|-|-
Description | If the user **does NOT have the Jetpack app installed**, tapping on the button should redirect the user to the Jetpack page in the App Store. | If the user **has the Jetpack app installed**, tapping on the button should open the Jetpack app.
Recording | <video src="https://user-images.githubusercontent.com/1299411/202095117-d8db5730-94dd-46d6-b51f-7e571974f48c.MP4"> | <video src="https://user-images.githubusercontent.com/1299411/202095015-8e3e45b1-8ce6-486f-a063-5b93d833590e.MP4">

FYI: @tiagomar and @zwarm

## To test

### Testing for scenario 1

- Ensure you don't have the Jetpack app on your test device.
- Build the WordPress app targeting your test device.
- Open the WordPress app.
- Ensure that the `Jetpack powered bottom sheet` flag is turned on.
- Tap on any Jetpack-powered label.
- Tap the "Try the new Jetpack app" button.
- 🔍 Verify that the WordPress app redirects you to Jetpack's App Store page.

### Testing for scenario 2

- **Don't delete the previously built WordPress app!**
- Ensure you don't have the Jetpack app on your test device.
- This time, build the Jetpack app targeting your test device.
- Open the WordPress app.
- Ensure that the `Jetpack powered bottom sheet` flag is turned on.
- Tap on any Jetpack-powered label.
- Tap the "Try the new Jetpack app" button.
- 🔍 Verify that the WordPress app now opens the Jetpack app.

## Regression Notes
1. Potential unintended areas of impact
Should be none.

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

4. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:
